src/UI/Internal/SelectionControl.elm

```diff
@@ -0,0 +1,69 @@
+module UI.Internal.SelectionControl exposing (..)
+
+import Element exposing (Attribute, px, shrink)
+import Element.Background as Background
+import Element.Border as Border
+import Html.Attributes as HtmlAttrs
+import UI.Internal.Colors as Colors
+
+
+type SelectionControlSize
+    = SizeSM
+    | SizeMD
+
+
+type alias Sizes =
+    { icon : Int, padding : Int, border : Int }
+
+
+iconColor : Bool -> Element.Color
+iconColor active =
+    if active then
+        Colors.primary.middle
+
+    else
+        Colors.gray.light1
+
+
+sizes : SelectionControlSize -> Sizes
+sizes size =
+    case size of
+        SizeSM ->
+            Sizes 20 8 2
+
+        SizeMD ->
+            Sizes 28 10 3
+
+
+buttonAttributes : SelectionControlSize -> List (Element.Attribute msg)
+buttonAttributes size =
+    [ Element.spacing 10
+    , Element.width shrink
```

Comment:
Please check what this looks like for both checkboxes and radio (better seen locally focusing on them) and let me know  whether you think the width should be configurable or not

Reply:
I liked the "shrink", I don't see in this combination (radio/check ++ text label) a reason for it to fill.
As this is just an "update" for already-existence implementation, we can go with shrink, and then for each new occurrence ask the UI team what should be the behavior...